python/dgl/nn/pytorch/conv/gatedgcnconv.py

```diff
@@ -0,0 +1,180 @@
+    residual : bool
+        Whether to include residual connection . Default: ``True``.
+    activation : callable activation function/layer or None, optional.
+        If not None, applies an activation function to the updated node features.
+        Default: ``None``.
+
+    Example
+    -------
+    >>> import dgl
+    >>> import torch as th
+    >>> import torch.nn.functional as F
+    >>> from dgl.nn import GatedGCNConv
+    >>> num_nodes, num_edges = 8, 30
+    >>> graph = dgl.rand_graph(num_nodes,num_edges)
+    >>> node_feats = th.rand((num_nodes, 20))
+    >>> edge_feats = th.rand((num_edges, 12))
+    >>> gatedGCN = GatedGCNConv(input_feats=20,
+    ...                         edge_feats=12,
+    ...                         output_feats=20,
+    ...                         dropout=0.2,
+    ...                         batch_norm=True,
+    ...                         residual=True,
+    ...                         activation=F.relu)
+    >>> new_node_feats, new_edge_feats = gatedGCN(graph, node_feats, edge_feats)
+    >>> new_node_feats.shape, new_edge_feats.shape
+    (torch.Size([8, 20]), torch.Size([30, 20]))
+
+    """
+
+    def __init__(
+        self,
+        input_feats,
+        edge_feats,
+        output_feats,
+        dropout=0,
+        batch_norm=True,
+        residual=True,
+        activation=F.relu,
+    ):
+        super(GatedGCNConv, self).__init__()
+        self._input_feats = input_feats
+        self._output_feats = output_feats
+        self.dropout = nn.Dropout(dropout)
+        self.batch_norm = batch_norm
+        self.residual = residual
+
+        if input_feats != output_feats or edge_feats != output_feats:
+            self.residual = False
+
+        # Linearly tranform the node features.
+        self.A = nn.Linear(input_feats, output_feats, bias=True)
+        self.B = nn.Linear(input_feats, output_feats, bias=True)
+        self.D = nn.Linear(input_feats, output_feats, bias=True)
+        self.E = nn.Linear(input_feats, output_feats, bias=True)
+
+        # Linearly tranform the edge features.
+        self.C = nn.Linear(edge_feats, output_feats, bias=True)
+
+        # Batch normalization on the node/edge features.
+        self.bn_node = nn.BatchNorm1d(output_feats)
+        self.bn_edge = nn.BatchNorm1d(output_feats)
+
+        self.activation = activation
+
+    def forward(self, graph, feat, edge_feat):
+        """
+
+        Description
+        -----------
+        Compute gated graph convolution layer.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            The graph.
+        feat : torch.Tensor
+            The input feature of shape :math:`(N, D_{in})` where :math:`N`
+            is the number of nodes of the graph and :math:`D_{in}` is the
+            input feature size.
+        edge_feat : torch.Tensor
+            The input edge feature of shape :math:`(E, D_{in_{edge}})`,
+            where :math:`E` is the number of edges and :math:`D_{in_{edge}}`
```

Comment:
Perhaps it's easier to use `D_{node}` in place of `D_{in}` above and `D_{edge}` here.